packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -424,6 +424,7 @@ class BackdropFilter extends SingleChildRenderObjectWidget {
   /// Creates a backdrop filter.
   ///
   /// The [filter] argument must not be null.
+  /// Backdrop filter will not apply when there is no [child] with a size.
```

Comment:
This contradicts some of the text above ("The filter will be applied to all the area within its parent or ancestor widget's clip. If there's no clip, the filter will be applied to the full screen."). Which is correct?

We should probably include some diagrams showing what happens (generated by code in the assets-for-api-docs repo so that we can prove that they are accurate!).

Reply:
I have been testing for some time. The main thing is that BackdropFilter needs the child to be rendered to apply its effect to the parent. So they are both correct, but if the child is not to be rendered (like a SizedBox, or a Container without a child and just width and height) the effect is never applied. How would you recommend to change the documentation?

Reply:
In looking through the code, particularly in RenderBackdropFilter, it will apply if the child isn't null. If the child is non-null, then it will apply regardless of size. Is something withholding the child from the RenderObject if it has no size?

Reply:
The problem appears to be in the engine where a layer with empty bounds will not be painted by the container recursion. In the case of BackdropLayer it will declare its paint bounds to be the bounds of the child, but it should probably declare its paint bounds to be the bounds of its backdrop (intersected with the clip).

Reply:
Interesting. How should we approach this? Should it be a documentation change or a code change like you suggested?

Reply:
I moved this discussion to the original issue.